modcc/cudaprinter.cpp

```diff
@@ -451,67 +463,70 @@ CUDAPrinter::CUDAPrinter(Module &m, bool o)
      *
      **************************************************************************/
 
-    // void set_ion(ionKind k, ion_type& i, const std::vector<size_type>&) override
-    //      TODO: this is done manually, which isn't going to scale
-    auto has_variable = [] (IonDep const& ion, std::string const& name) {
-        if( std::find_if(ion.read.begin(), ion.read.end(),
-                      [&name] (Token const& t) {return t.spelling==name;}
-            ) != ion.read.end()
-        ) return true;
-        if( std::find_if(ion.write.begin(), ion.write.end(),
-                      [&name] (Token const& t) {return t.spelling==name;}
-            ) != ion.write.end()
-        ) return true;
-        return false;
+    // return true/false indicating if cell has dependency on k
+    auto const& ions = module_->neuron_block().ions;
+    auto find_ion = [&ions] (ionKind k) {
+        return std::find_if(
+            ions.begin(), ions.end(),
+            [k](IonDep const& d) {return d.kind()==k;}
+        );
+    };
+    auto has_ion = [&ions, find_ion] (ionKind k) {
+        return find_ion(k) != ions.end();
```

Comment:
Common code here with `CPrinter` could be moved to the `Module` interface. Would it simplify the interface and code generation if the `Module` could provide the sequence of `IonDep` ion descriptions?

Reply:
Good point. I moved it to `Module`